lib/spotify/song-update.ts

```diff
@@ -0,0 +1,38 @@
+import fetch from 'node-fetch';
+import { isTopSongsResponse, TopSongsResponse } from './types';
+
+export async function getTopSongs(accessToken: string): Promise<TopSongsResponse>{
+  function getSongs(offset: number){
+    return fetch(`https://api.spotify.com/v1/me/top/tracks?limit=50&time_range=medium_term&offset=${offset}`, {
+    method: 'GET',
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+    },
+    });
+  }
+
+  const responses = await Promise.all([getSongs(0), getSongs(49)])
+  const data = await Promise.all(responses.map(res => res.json()));
+  const extractedItems = data.reduce((all: any[], d) => {
+    all.push(...d.items);
+    return all;
+  }, []);
```

Comment:
this reducer doesn't need to exist.. i think we can have two nested reducers to do both `extractedItems` and `songMap`

Basically, find a way to simplify this to optimize